decidim-census/app/controllers/decidim/census/admin/censuses_controller.rb

```diff
@@ -0,0 +1,45 @@
+
+module Decidim
+  module Census
+    module Admin
+      class CensusesController < Decidim::Admin::ApplicationController
+
+        before_action :show_instructions, unless: :census_authorization_active_in_organization?
+
+        def show
+          authorize! :show, CensusDatum
+          @status = Status.new(current_organization)
+        end
+
+        def create
+          authorize! :create, CensusDatum
+          if params[:file]
```

Comment:
This implementation is correct. However, I would have used a more Rails like approach to the problem.

```ruby

def create
   authorize! :create, CensusDatum
   csv_data = CsvData.new(file: params[:file])
   if csv_file.valid?
      csv_data.parse
      CensusDatum.insert_all(current_organization, csv_data.values)
      RemoveDuplicatesJob.perform_later(current_organization)
      flash[:notice] = t('.success', count: data.values.count,
                                   errors: data.errors.count)
   end
   redirect_to censuses_path
end

```

And then, In CsvData:

```ruby
class CsvData
  include ActiveModel::Model
  attr_accessor :file

  validates :file, presence: true
end
```

However, I don't believe that we should change the current code, because adding ActiveModel here is overkill for what we need. I'm making this comment as an example of alternate implementations.

Reply:
Great! I'll think about it